fastapi/encoders.py

```diff
@@ -105,24 +105,22 @@ def jsonable_encoder(
                 )
             )
         return encoded_list
+    for encoder_type, encoder in custom_encoder.items():
```

Comment:
I did a check, and `if custom_encoder:` is ~3-4x faster than `for encoder_type, encoder in custom_encoder.items():` when custom_encoder is empty. So I think this branch should be behind an `if custom_encoder:` check.

(Note that this function calls itself recursively for lists and dicts, so should be considered a hot-loop function, and should be implemented as performantly as possible.)

Reply:
Well you said earlier: https://github.com/tiangolo/fastapi/pull/756#issuecomment-560042441 that since this is only for custom encoders it wasn't too important, that's why I didn't bother.

I can definitely do some performance testing and speed it up significantly with your suggestion though!

Reply:
The thing that concerns me is that there are some relatively commonly used types in here:
https://github.com/samuelcolvin/pydantic/blob/31cac4783b2cba2836e7bf2b4a593a3085930da6/pydantic/json.py#L20

At the very least, `Decimal` is somewhat common, and it is at the end of the list. Either way, it's probably worth a PR to pydantic to reorder based on most common usage. But that can be separate.